library/bignum.c

```diff
@@ -588,7 +604,7 @@ int mbedtls_mpi_write_string( const mbedtls_mpi *X, int radix,
         if( T.s == -1 )
```

Comment:
Also, if we want to safe code, I think we should call `mpi_write_hlp()` even for radix `16` and get rid of the `radix == 16` branch - it might not be as fast as the hand-coded version, but `mbedtls_mpi_write_string()` is not time-critical anyway.

Reply:
We could do that, but I have no strong opinions about it, so this is not a blocker for my approval...